Core/include/Acts/EventData/ParameterSet.hpp

```diff
@@ -578,29 +402,37 @@ class ParameterSet {
    * @param values vector with parameter values to be checked and corrected if
    * necessary
    */
-  static void correctValues(ParameterVector& values) {
-    detail::value_corrector<params...>::result(values);
+  static void correctValues(ParametersVector& values) {
+    detail::ValueCorrector<parameter_indices_t, params...>::run(values);
   }
 
  private:
-  ParameterVector m_vValues{
-      ParameterVector::Zero()};  ///< column vector containing
-                                 ///< values of local parameters
-  std::optional<CovarianceMatrix> m_optCovariance{
-      std::nullopt};  ///< an optional covariance matrix
-
-  static const Projection sProjector;  ///< matrix to project full parameter
-                                       /// vector onto local parameter space
+  /// column vector containing values of local parameters.
+  ParametersVector m_vValues{ParametersVector::Zero()};
+  /// optional covariance matrix.
+  std::optional<CovarianceMatrix> m_optCovariance{std::nullopt};
+
+  /// matrix to project full parameter vector onto local parameter space.
+  static const ProjectionMatrix sProjector;
+
+  /// Compare two parameter sets for value equality.
+  friend constexpr bool operator==(const ParameterSet& lhs,
```

Comment:
Why does this have to be friend? And it's definitely not actually constexpr, so maybe don't call it that.

Reply:
It accesses private member and thus has to be a `friend`. I can remove the `constexpr`.

Reply:
But it's a method, so should have access to the member anyway?

Reply:
It is not a method. By using `friend` it becomes a friend function.

Reply:
But it can just be a method, no need for `friend` then. No?

Reply:
```cpp
bool operator==(const ParameterSet& rhs) const {
   return (m_vValues == rhs.m_vValues) and
           (m_optCovariance == rhs.m_optCovariance);
}
```

Reply:
Yes, it can but I prefer the `friend` variant as it treats both operands as equal. If you write
```cpp
a == b
```
I makes more sense to me to interpret this as a binary operator `equal(a, b)` instead of `a.equal(b)`.

Reply:
Hm ok. I think `friend` should be avoided if **at all** possible. It's one of the worst warts of the language.

Reply:
It does not have to be `friend`. It could be a regular free function. Using `friend` just allows a much simpler implementation.

Since the functionality is the same, is there a technical reason to switch this?

Reply:
`friend` is bad.

Reply:
Just the fact that `friend` magically makes this a function that is not a member (no `this`), but its declared and defined inside the class body should give you chills of terror down your spine and make you not want to touch it without gloves on.

Reply:
😆 I think we have quite different reactions to this code. This might not be a technical reason, but it is enough to change it.

Reply:
Fixed.